parser-library/buffer.cpp

```diff
@@ -23,47 +23,52 @@ THE SOFTWARE.
 */
 
 #include "parse.h"
+
+#include <cstring>
 #include <fstream>
-#include <string.h>
 
 #ifdef WIN32
+#include <intrin.h>
 #include <windows.h>
 #else
 #include <fcntl.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
-#include <sys/types.h>
 #include <unistd.h>
 #endif
 
 namespace {
 
-inline uint16_t byteSwapUint16(uint16_t val) {
-  uint16_t a = (val >> 8) & 0x00FFU;
-  uint16_t b = (val << 8) & 0xFF00U;
-  return a | b;
+inline std::uint16_t byteSwapUint16(std::uint16_t val) {
```

Comment:
These functions were triggering some warnings; I am now using built-ins to handle swapping.